fedimint-client/src/lib.rs

```diff
@@ -2145,32 +2177,34 @@ impl ClientBuilder {
                         let final_client = final_client.clone();
                         let (progress_tx, progress_rx) = tokio::sync::watch::channel(progress);
                         let module_init = module_init.clone();
+                        let meta_manager = self.meta_manager.clone();
                         (
                             Box::pin(async move {
                                 module_init
-                                        .recover(
-                                            final_client.clone(),
-                                            fed_id,
+                                    .recover(
+                                        final_client.clone(),
+                                        fed_id,
                                         num_peers,
-                                            module_config.clone(),
-                                            db.clone(),
-                                            module_instance_id,
-                                            common_api_versions.core,
-                                            api_version,
-                                            root_secret.derive_module_secret(module_instance_id),
-                                            notifier.clone(),
-                                            api.clone(),
+                                        module_config.clone(),
+                                        db.clone(),
+                                        module_instance_id,
+                                        common_api_versions.core,
+                                        api_version,
+                                        root_secret.derive_module_secret(module_instance_id),
+                                        notifier.clone(),
+                                        api.clone(),
                                         admin_auth,
-                                            snapshot.as_ref().and_then(|s| s.modules.get(&module_instance_id).to_owned()),
-                                            progress_tx,
-                                        )
-                                        .await
-                                        .map_err(|err| {
-                                            warn!(
-                                                module_id = module_instance_id, %kind, %err, "Module failed to recover"
-                                            );
-                                            err
-                                        })
+                                        snapshot.as_ref().and_then(|s| s.modules.get(&module_instance_id).to_owned()),
+                                        progress_tx,
+                                        meta_manager,
```

Comment:
"meta" was supposed to be extra functionality for end apps, why would models now need to depend on it?

Reply:
That whole LN depending on Meta seems like a potential mistake. I thought we already moved active gateway out of client to the user, should vetted gateways be kind of the same?

Also - if we do want to reuse meta for it, maybe we should use a key `1` for "LN-meta" as opposed to default `0`, so they can be voted, changed and fetched independently?

Reply:
fixing it slightly in https://github.com/fedimint/fedimint/pull/4785

now with meta manager, it should be much easier for client impls.

Reply:
removed accesss to meta manager from module args